core/includes/drupal.inc

```diff
@@ -889,5 +889,39 @@ function system_settings_form_submit($form, &$form_state) {
 }
 
 /**
+ * Alters Backdrop's list of defined date formats and attributes to return the
+ * expected date types and attributes for Drupal's system_get_date_types()
+ * function.
+ *
+ * @deprecated since 1.0
+ */
+function system_get_date_types($type = NULL) {
+  $formats = system_get_date_formats($type);
+
+  if ($type) {
```

Comment:
Hi @davereid! I think you're right. And goodness this function's logic is confusing to follow! Let's open a new issue to fix it.